### PR TITLE
feat(service-course, service-food, service-news): support pagination

### LINKED ISSUE #26: Support pagination
**Description**

### PR DESCRIPTION
**Pull request by JonasWanke**
**Fixes: #26**

**Checklist**

- [x] Commit messages follow conventionalcommits.org
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Changes**

### CODE REVIEW

File: common/src/main/kotlin/de/hpi/cloud/common/utils/couchbase/N1ql.kt [hunk 1]
```diff
@@ -25,4 +25,8 @@ fun descTimestamp(field: Expression): Array<Sort> {
 
 fun n(vararg part: String): Expression {
     return x(part.joinToString(NESTED_SEPARATOR.toString()) { i(it).toString() })
-}
\ No newline at end of file
+}
+
+fun v(vararg part: String): Expression {
```
Comment (ShirleyNekoDev) [inline common/src/main/kotlin/de/hpi/cloud/common/utils/couchbase/N1ql.kt]: Bitte Javadocs für diese Funktionen

File: service-course/src/main/kotlin/de/hpi/cloud/course/CourseService.kt [hunk 2]
```diff
@@ -110,26 +111,23 @@ class CourseServiceImpl(private val bucket: Bucket) : CourseServiceGrpc.CourseSe
...
-                }
-
-            ListCoursesResponse.newBuilder()
-                .addAllCourses(courses.map { it.parseCourse() })
-                .build()
+            val (courses, newToken) = paginate(bucket, {
+                where(
+                    and(
+                        x(KEY_TYPE).eq(s("course")),
+                        v("courseSeriesId").eq(s(courseSeriesId)).takeIf { courseSeriesId != null },
```
Comment (ShirleyNekoDev) [inline service-course/src/main/kotlin/de/hpi/cloud/course/CourseService.kt:118]: Ich habe keine Ahnung, was das machen soll.
Vielleicht passendere Namen für die Funktionen "x,v,s" finden.
Ggf. manche als Extension auf String hinzufügen?

File: service-food/src/main/kotlin/de/hpi/cloud/food/FoodService.kt [hunk 3]
```diff
@@ -68,34 +68,29 @@ class FoodServiceImpl(private val bucket: Bucket) : FoodServiceGrpc.FoodServiceI
...
-            .build()
+        val (menuItems, newToken) = paginate(bucket, {
+            where(
+                and(
+                    x(KEY_TYPE).eq(s("menuItem")),
+                    restaurantId?.let { n(KEY_VALUE, "restaurantId").eq(s(restaurantId)) },
+                    date?.let {
+                        millisToStr(
+                            n(KEY_VALUE, "date", "millis"),
+                            "1111-11-11"
```
Comment (ShirleyNekoDev) [inline service-food/src/main/kotlin/de/hpi/cloud/food/FoodService.kt]: Kein Plan, was die Query machen soll...
Wofür steht "1111-11-11"?

**Review by ShirleyNekoDev — CHANGES_REQUESTED** [review]

File: service-course/src/main/kotlin/de/hpi/cloud/course/CourseService.kt (hunk 2 above)
Comment (JonasWanke) [reply service-course/src/main/kotlin/de/hpi/cloud/course/CourseService.kt:118]: This is a N1ql query that was already there before this PR. The functions `x` and `s` are provided by Couchbase to build a query with proper escaping (second last paragraph of <https://docs.couchbase.com/java-sdk/2.7/n1ql-queries-with-sdk.html#building-statements-with-the-dsl>), `v` is just a shorthand introduced by us. This has to do with the general design of the Couchbase Java SDK and is not part of this PR.

File: service-food/src/main/kotlin/de/hpi/cloud/food/FoodService.kt (hunk 3 above)
Comment (JonasWanke) [reply service-food/src/main/kotlin/de/hpi/cloud/food/FoodService.kt]: Again, this was already here before this PR and is documented in the Couchbase documentation: <https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/datefun.html#fn-date-millis-to-str>

File: common/src/main/kotlin/de/hpi/cloud/common/utils/couchbase/N1ql.kt (hunk 1 above)
Comment (JonasWanke) [reply common/src/main/kotlin/de/hpi/cloud/common/utils/couchbase/N1ql.kt]: **KDoc** was added

**Review by ShirleyNekoDev — APPROVED** [review]

**Pull request merged**